setup.py

```diff
@@ -41,13 +43,41 @@
     define_macros=[("SOURCE_PATH_SIZE", str(len(os.path.abspath("./src/clp/components/core"))))],
 )
 
+def _parallel_compile(
+    self: distutils.ccompiler.CCompiler,
+    sources: List[str],
+    output_dir: Optional[str] = None,
+    macros: Optional[List[Tuple[str, Optional[str]]]] = None,
+    include_dirs: Optional[List[str]] = None,
+    debug: int = 0,
+    extra_preargs: Optional[List[str]] = None,
+    extra_postargs: Optional[List[str]] = None,
+    depends: Optional[List[str]] = None,
+) -> List[str]:
+    macros, objects, extra_postargs, pp_opts, build = self._setup_compile(
+        output_dir, macros, include_dirs, sources, depends, extra_postargs
+    )
+    cc_args = self._get_cc_args(pp_opts, debug, extra_preargs)
+
+    def _compile_single_file(obj: str) -> None:
+        try:
+            src, ext = build[obj]
+        except KeyError:
+            return
+        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+
+    num_cores: int = multiprocessing.cpu_count()
+    multiprocessing.pool.ThreadPool(num_cores).map(_compile_single_file, objects)
```

Comment:
We should use `ThreadPool` with `with`, right (the docs say it should be [closed](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.ThreadPool))?

Reply:
right, mb